src/main/java/sparksoniq/spark/iterator/flowr/GroupByClauseSparkIterator.java

```diff
@@ -280,81 +280,66 @@ private void linearizeTuples(List<FlworTuple> keyTuplePairs) {
         columnNames = Arrays.asList(columnNamesArray);
 
         df.createOrReplaceTempView("input");
-        df.sparkSession().table("input").cache();
-        df.sparkSession().udf().register("determineGroupingDataType",
-                new GroupClauseDetermineTypeUDF(variableAccessExpressions, columnNames),
-                DataTypes.createArrayType(DataTypes.StringType));
-
-        String udfSQL = DataFrameUtils.getSQL(inputSchema, -1, false);
-
-        Dataset<Row> columnTypesDf = df.sparkSession().sql(
-                String.format("select distinct(determineGroupingDataType(array(%s))) as `distinct-types` from input",
-                        udfSQL)
-        );
-        Object columnTypesObject = columnTypesDf.collect();
-        Row[] columnTypesOfRows = ((Row[]) columnTypesObject);
+//        df.sparkSession().table("input").cache();
```

Comment:
Just as a reminder, let's not forget to remove the commented out code before merging.